src/attestedclaim/AttestedClaim.ts

```diff
@@ -47,18 +61,25 @@ export default class AttestedClaim implements IAttestedClaim {
    *
    * @param request - A request for attestation, usually sent by a claimer.
    * @param attestation - The attestation to base the [[AttestedClaim]] on.
-   * @example Create an [[AttestedClaim]] upon successful [[Attestation]] creation:
-   * ```javascript
+   * @example ```javascript
+   * //Create an [[AttestedClaim]] upon successful [[Attestation]] creation:
    * new AttestedClaim(requestForAttestation, attestation);
    * ```
    */
-  public constructor(
-    request: IRequestForAttestation,
-    attestation: IAttestation
-  ) {
-    // TODO: this should be instantiated w/o fromObject
-    this.request = RequestForAttestation.fromObject(request)
-    this.attestation = Attestation.fromObject(attestation)
+  public constructor(attestedClaimInput: IAttestedClaim) {
+    if (!attestedClaimInput.request || !attestedClaimInput.attestation) {
+      throw new Error(
+        `Property Not Provided while building AttestedClaim!\n
```

Comment:
Your error messages are really helpful, good one.

Suggestion (not for this PR so that we keep moving, but as a follow-up): create a ticket to move this validation logic and errors to a separate function in order to make the constructor less verbose. Same for all constructors where you're doing something similar.

Reply:
good point!